python/lsst/ts/simactuators/tracking_actuator.py

```diff
@@ -0,0 +1,209 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["TrackingActuator"]
+
+from lsst.ts import salobj
+
+from . import path
+
+
+class TrackingActuator:
+    """Simulate an actuator that slews to and tracks a path defined
+    by regular position, velocity, time updates.
+
+    Parameters
+    ----------
+    min_pos : `float`
+        Minimum allowed position (deg)
+    max_pos : `float`
+        Maximum allowed position (deg)
+    max_vel : `float`
+        Maximum allowed velocity (deg/sec)
+    max_accel : `float`
+        Maximum allowed acceleration (deg/sec^2)
+    dtmax_track : `float`
+        Maximum allowed time interval (t - time of last segment)
+        for `set_cmd` to compute a tracking path (sec); if this limit
+        is not met then `set_cmd` computes a slewing path.
+        This should be larger than the maximum expected time between calls to
+        `set_cmd`, but not much more than that.
+    nsettle : `int` (optional)
+        Number of calls to `set_cmd` after a slew finishes
+        (meaning ``self.curr.kind`` is tracking)
+        before ``self.kind(t)`` reports tracking instead of slewing.
+    t : `float` (optional)
+        Time (TAI unix seconds, e.g. from lsst.ts.salobj.curr_tai())
+        for initial `cmd` `PathSegment` and `curr` `Path`.
+        If None then use current TAI.
+        This is primarily for unit tests; None is usually what you want.
+
+    Raises
+    ------
+    ValueError
+        If ``min_pos >= max_pos``, ``max_vel <= 0``, or ``max_accel <= 0``.
+    """
+    Kind = path.Kind
+
+    def __init__(self, min_pos, max_pos, max_vel, max_accel, dtmax_track, nsettle=2, start_time=None):
+        if min_pos >= max_pos:
+            raise ValueError(f"min_pos={min_pos} must be < max_pos={max_pos}")
+        if max_vel <= 0:
+            raise ValueError(f"max_vel={max_vel} must be > 0")
+        if max_accel <= 0:
+            raise ValueError(f"max_vel={max_vel} must be > 0")
+        self.min_pos = min_pos
+        self.max_pos = max_pos
+        self.max_vel = max_vel
+        self.max_accel = max_accel
+        self.dtmax_track = dtmax_track
+        self.nsettle = nsettle
+
+        if start_time is None:
+            start_time = salobj.current_tai()
+        if min_pos <= 0 and 0 < max_pos:
+            start_pos = 0
+        else:
+            start_pos = min_pos
+        self.cmd = path.PathSegment(start_time=start_time, start_pos=start_pos)
+        self.curr = path.Path(path.PathSegment(start_time=start_time, start_pos=start_pos),
```

Comment:
In the abstract I agree. cmd_path and curr_path would be nice names if only cmd_path was a path instead of a path segment. cmd_path_segment is too verbose. I have not yet thought of names I like better than cmd and curr.